make check in skeletons fails to build on Mac

macOS 10.12.6, Xcode 9.0. `make check` fails in the skeletons dir on the master branch.

`ar cru` works on libasn1cskeletons_c89.a but then fails on libasn1cskeletons_c89_32.a because no archive members are specified.

```
/bin/sh ../libtool  --tag=CC   --mode=link gcc -std=c89 -Wpedantic -Wno-long-long  -fsanitize=undefined -fno-sanitize-recover=undefined -fsanitize=unsigned-integer-overflow -fno-sanitize
-recover=unsigned-integer-overflow -fsanitize=nullability -fno-sanitize-recover=nullability -fno-omit-frame-pointer -fsanitize=address -fno-omit-frame-pointer -g -O2 -Wall -Wcast-qual -W
char-subscripts -Wmissing-prototypes -Wmissing-declarations -Wno-error=attributes -Wno-error=cast-align -Wno-error=visibility -Wno-error=parentheses-equality -std=gnu99   -o libasn1cskel
etons_c89.la  libasn1cskeletons_c89_la-ANY.lo libasn1cskeletons_c89_la-BIT_STRING.lo libasn1cskeletons_c89_la-BIT_STRING_oer.lo libasn1cskeletons_c89_la-BMPString.lo libasn1cskeletons_c8
9_la-BOOLEAN.lo libasn1cskeletons_c89_la-ENUMERATED.lo libasn1cskeletons_c89_la-GeneralString.lo libasn1cskeletons_c89_la-GeneralizedTime.lo libasn1cskeletons_c89_la-GraphicString.lo lib
asn1cskeletons_c89_la-IA5String.lo libasn1cskeletons_c89_la-INTEGER.lo libasn1cskeletons_c89_la-INTEGER_oer.lo libasn1cskeletons_c89_la-ISO646String.lo libasn1cskeletons_c89_la-NULL.lo l
ibasn1cskeletons_c89_la-NativeEnumerated.lo libasn1cskeletons_c89_la-NativeEnumerated_oer.lo libasn1cskeletons_c89_la-NativeInteger.lo libasn1cskeletons_c89_la-NativeInteger_oer.lo libas
n1cskeletons_c89_la-NativeReal.lo libasn1cskeletons_c89_la-NumericString.lo libasn1cskeletons_c89_la-OBJECT_IDENTIFIER.lo libasn1cskeletons_c89_la-OCTET_STRING.lo libasn1cskeletons_c89_l
a-OCTET_STRING_oer.lo libasn1cskeletons_c89_la-OPEN_TYPE.lo libasn1cskeletons_c89_la-OPEN_TYPE_oer.lo libasn1cskeletons_c89_la-ObjectDescriptor.lo libasn1cskeletons_c89_la-PrintableStrin
g.lo libasn1cskeletons_c89_la-REAL.lo libasn1cskeletons_c89_la-RELATIVE-OID.lo libasn1cskeletons_c89_la-T61String.lo libasn1cskeletons_c89_la-TeletexString.lo libasn1cskeletons_c89_la-UT
CTime.lo libasn1cskeletons_c89_la-UTF8String.lo libasn1cskeletons_c89_la-UniversalString.lo libasn1cskeletons_c89_la-VideotexString.lo libasn1cskeletons_c89_la-VisibleString.lo libasn1cs
keletons_c89_la-asn_SEQUENCE_OF.lo libasn1cskeletons_c89_la-asn_SET_OF.lo libasn1cskeletons_c89_la-asn_application.lo libasn1cskeletons_c89_la-asn_codecs_prim.lo libasn1cskeletons_c89_la
-asn_internal.lo libasn1cskeletons_c89_la-asn_bit_data.lo libasn1cskeletons_c89_la-asn_random_fill.lo libasn1cskeletons_c89_la-ber_decoder.lo libasn1cskeletons_c89_la-ber_tlv_length.lo l
ibasn1cskeletons_c89_la-ber_tlv_tag.lo libasn1cskeletons_c89_la-constr_CHOICE.lo libasn1cskeletons_c89_la-constr_CHOICE_oer.lo libasn1cskeletons_c89_la-constr_SEQUENCE.lo libasn1cskeleto
ns_c89_la-constr_SEQUENCE_oer.lo libasn1cskeletons_c89_la-constr_SEQUENCE_OF.lo libasn1cskeletons_c89_la-constr_SET.lo libasn1cskeletons_c89_la-constr_SET_OF.lo libasn1cskeletons_c89_la-
constr_SET_OF_oer.lo libasn1cskeletons_c89_la-constr_TYPE.lo libasn1cskeletons_c89_la-constraints.lo libasn1cskeletons_c89_la-der_encoder.lo libasn1cskeletons_c89_la-oer_decoder.lo libas
n1cskeletons_c89_la-oer_encoder.lo libasn1cskeletons_c89_la-oer_support.lo libasn1cskeletons_c89_la-per_decoder.lo libasn1cskeletons_c89_la-per_encoder.lo libasn1cskeletons_c89_la-per_op
entype.lo libasn1cskeletons_c89_la-per_support.lo libasn1cskeletons_c89_la-xer_decoder.lo libasn1cskeletons_c89_la-xer_encoder.lo libasn1cskeletons_c89_la-xer_support.lo  
libtool: link: ar cru .libs/libasn1cskeletons_c89.a .libs/libasn1cskeletons_c89_la-ANY.o .libs/libasn1cskeletons_c89_la-BIT_STRING.o .libs/libasn1cskeletons_c89_la-BIT_STRING_oer.o .libs/libasn1cskeletons_c89_la-BMPString.o .libs/libasn1cskeletons_c89_la-BOOLEAN.o .libs/libasn1cskeletons_c89_la-ENUMERATED.o .libs/libasn1cskeletons_c89_la-GeneralString.o .libs/libasn1cskeletons_c89_la-GeneralizedTime.o .libs/libasn1cskeletons_c89_la-GraphicString.o .libs/libasn1cskeletons_c89_la-IA5String.o .libs/libasn1cskeletons_c89_la-INTEGER.o .libs/libasn1cskeletons_c89_la-INTEGER_oer.o .libs/libasn1cskeletons_c89_la-ISO646String.o .libs/libasn1cskeletons_c89_la-NULL.o .libs/libasn1cskeletons_c89_la-NativeEnumerated.o .libs/libasn1cskeletons_c89_la-NativeEnumerated_oer.o .libs/libasn1cskeletons_c89_la-NativeInteger.o .libs/libasn1cskeletons_c89_la-NativeInteger_oer.o .libs/libasn1cskeletons_c89_la-NativeReal.o .libs/libasn1cskeletons_c89_la-NumericString.o .libs/libasn1cskeletons_c89_la-OBJECT_IDENTIFIER.o .libs/libasn1cskeletons_c89_la-OCTET_STRING.o .libs/libasn1cskeletons_c89_la-OCTET_STRING_oer.o .libs/libasn1cskeletons_c89_la-OPEN_TYPE.o .libs/libasn1cskeletons_c89_la-OPEN_TYPE_oer.o .libs/libasn1cskeletons_c89_la-ObjectDescriptor.o .libs/libasn1cskeletons_c89_la-PrintableString.o .libs/libasn1cskeletons_c89_la-REAL.o .libs/libasn1cskeletons_c89_la-RELATIVE-OID.o .libs/libasn1cskeletons_c89_la-T61String.o .libs/libasn1cskeletons_c89_la-TeletexString.o .libs/libasn1cskeletons_c89_la-UTCTime.o .libs/libasn1cskeletons_c89_la-UTF8String.o .libs/libasn1cskeletons_c89_la-UniversalString.o .libs/libasn1cskeletons_c89_la-VideotexString.o .libs/libasn1cskeletons_c89_la-VisibleString.o .libs/libasn1cskeletons_c89_la-asn_SEQUENCE_OF.o .libs/libasn1cskeletons_c89_la-asn_SET_OF.o .libs/libasn1cskeletons_c89_la-asn_application.o .libs/libasn1cskeletons_c89_la-asn_codecs_prim.o .libs/libasn1cskeletons_c89_la-asn_internal.o .libs/libasn1cskeletons_c89_la-asn_bit_data.o .libs/libasn1cskeletons_c89_la-asn_random_fill.o .libs/libasn1cskeletons_c89_la-ber_decoder.o .libs/libasn1cskeletons_c89_la-ber_tlv_length.o .libs/libasn1cskeletons_c89_la-ber_tlv_tag.o .libs/libasn1cskeletons_c89_la-constr_CHOICE.o .libs/libasn1cskeletons_c89_la-constr_CHOICE_oer.o .libs/libasn1cskeletons_c89_la-constr_SEQUENCE.o .libs/libasn1cskeletons_c89_la-constr_SEQUENCE_oer.o .libs/libasn1cskeletons_c89_la-constr_SEQUENCE_OF.o .libs/libasn1cskeletons_c89_la-constr_SET.o .libs/libasn1cskeletons_c89_la-constr_SET_OF.o .libs/libasn1cskeletons_c89_la-constr_SET_OF_oer.o .libs/libasn1cskeletons_c89_la-constr_TYPE.o .libs/libasn1cskeletons_c89_la-constraints.o .libs/libasn1cskeletons_c89_la-der_encoder.o .libs/libasn1cskeletons_c89_la-oer_decoder.o .libs/libasn1cskeletons_c89_la-oer_encoder.o .libs/libasn1cskeletons_c89_la-oer_support.o .libs/libasn1cskeletons_c89_la-per_decoder.o .libs/libasn1cskeletons_c89_la-per_encoder.o .libs/libasn1cskeletons_c89_la-per_opentype.o .libs/libasn1cskeletons_c89_la-per_support.o .libs/libasn1cskeletons_c89_la-xer_decoder.o .libs/libasn1cskeletons_c89_la-xer_encoder.o .libs/libasn1cskeletons_c89_la-xer_support.o 
libtool: link: ranlib .libs/libasn1cskeletons_c89.a
libtool: link: ( cd ".libs" && rm -f "libasn1cskeletons_c89.la" && ln -s "../libasn1cskeletons_c89.la" "libasn1cskeletons_c89.la" )
/bin/sh ../libtool  --tag=CC   --mode=link gcc  -g -O2 -Wall -Wcast-qual -Wchar-subscripts -Wmissing-prototypes -Wmissing-declarations -Wno-error=attributes -Wno-error=cast-align -Wno-error=visibility -Wno-error=parentheses-equality -std=gnu99   -o libasn1cskeletons_c89_32.la    
libtool: link: ar cru .libs/libasn1cskeletons_c89_32.a 
ar: no archive members specified
usage:  ar -d [-TLsv] archive file ...
        ar -m [-TLsv] archive file ...
        ar -m [-abiTLsv] position archive file ...
        ar -p [-TLsv] archive [file ...]
        ar -q [-cTLsv] archive file ...
        ar -r [-cuTLsv] archive file ...
        ar -r [-cuTLsv] archive file ...
        ar -r [-abciuTLsv] position archive file ...
        ar -t [-TLsv] archive [file ...]
        ar -x [-ouTLsv] archive [file ...]
make[1]: *** [libasn1cskeletons_c89_32.la] Error 1
make: *** [check-am] Error 2
```

I've tried the current master with and without #229, with and without `--disable-test-32bit`. Two tests inevitably fail:
```
. . . . .
============================================================================
Testsuite summary for asn1c 0.9.29
============================================================================
# TOTAL: 40
# PASS:  40
# SKIP:  0
# XFAIL: 0
# FAIL:  0
# XPASS: 0
# ERROR: 0
============================================================================
Making check in tests-randomized
/Applications/Xcode.app/Contents/Developer/usr/bin/make  check-bundles.sh
make[3]: Nothing to be done for `check-bundles.sh'.
/Applications/Xcode.app/Contents/Developer/usr/bin/make  check-TESTS
PASS: bundles/00-NULL-bundle.txt
PASS: bundles/01-BOOLEAN-bundle.txt
PASS: bundles/02-INTEGER-bundle.txt
PASS: bundles/03-ENUMERATED-bundle.txt
PASS: bundles/04-REAL-bundle.txt
PASS: bundles/05-BIT-STRING-bundle.txt
PASS: bundles/06-OCTET-STRING-bundle.txt
PASS: bundles/07-VisibleString-bundle.txt
PASS: bundles/08-OBJECT-IDENTIFIER-bundle.txt
PASS: bundles/09-RELATIVE-OID-bundle.txt
PASS: bundles/10-UTF8String-bundle.txt
PASS: bundles/11-BMPString-bundle.txt
PASS: bundles/12-UniversalString-bundle.txt
PASS: bundles/13-UTCTime-bundle.txt
PASS: bundles/14-GeneralizedTime-bundle.txt
PASS: bundles/15-CHOICE-bundle.txt
PASS: bundles/16-SEQUENCE-bundle.txt
FAIL: bundles/17-SEQUENCE-OF-bundle.txt
FAIL: bundles/19-SET-OF-bundle.txt
============================================================================
Testsuite summary for asn1c 0.9.29
============================================================================
# TOTAL: 19
# PASS:  17
# SKIP:  0
# XFAIL: 0
# FAIL:  2
# XPASS: 0
# ERROR: 0
============================================================================
See tests/tests-randomized/test-suite.log
Please report to vlm@lionet.info
============================================================================
make[4]: *** [test-suite.log] Error 1
make[3]: *** [check-TESTS] Error 2
make[2]: *** [check-am] Error 2
make[1]: *** [check-recursive] Error 1
make: *** [check-recursive] Error 1
```

Here's the `tests/tests-randomized/test-suite.log` file: 
[randomized-test-suite.log.txt](https://github.com/vlm/asn1c/files/1415959/randomized-test-suite.log.txt)

--- a/skeletons/Makefile.am
+++ b/skeletons/Makefile.am
@@ -20,13 +20,13 @@ uninstall-local:
 
 check_LTLIBRARIES =             \
     libasn1cskeletons.la        \
-    libasn1cskeletons_c89.la    \
-    libasn1cskeletons_c89_32.la
+    libasn1cskeletons_c89.la
 
 libasn1cskeletons_c89_la_CFLAGS = $(SKELETONS_CFLAGS)
 libasn1cskeletons_c89_la_SOURCES = $(libasn1cskeletons_la_SOURCES)
 
 if EXPLICIT_M32
+check_LTLIBRARIES += libasn1cskeletons_c89_32.la
 libasn1cskeletons_c89_32_la_CFLAGS = $(SKELETONS_CFLAGS) $(CFLAGS_M32) -Wno-format
 libasn1cskeletons_c89_32_la_SOURCES = $(libasn1cskeletons_la_SOURCES)
 endif
@@ -103,8 +103,7 @@ libasn1cskeletons_la_SOURCES =                  \
 
 check_PROGRAMS =                    \
     check-converter_example         \
-    check-converter_c89_example     \
-    check-converter_c89_32_example
+    check-converter_c89_example
 LDADD = -lm
 
 check_converter_example_CFLAGS = -DNO_ASN_PDU
@@ -115,7 +114,10 @@ check_converter_c89_example_CFLAGS = $(SKELETON_CFLAGS) -DNO_ASN_PDU
 check_converter_c89_example_SOURCES = converter-example.c
 check_converter_c89_example_LDADD = libasn1cskeletons_c89.la
 
+if EXPLICIT_M32
+check_PROGRAMS += check-converter_c89_32_example
 check_converter_c89_32_example_CFLAGS = $(SKELETON_CFLAGS) $(CFLAGS_M32) -DNO_ASN_PDU
 check_converter_c89_32_example_SOURCES = converter-example.c
 check_converter_c89_32_example_LDADD = libasn1cskeletons_c89_32.la
+endif
 

--- a/skeletons/constr_SEQUENCE_OF.c
+++ b/skeletons/constr_SEQUENCE_OF.c
@@ -183,6 +183,14 @@ SEQUENCE_OF_encode_uper(const asn_TYPE_descriptor_t *td,
                                 ct->effective_bits))
                 ASN__ENCODE_FAILED;
         }
+    } else if (list->count == 0) {
+        /* When the list is empty add only the length determinant
+         * X.691, #20.6 and #11.9.4.1
+         */
+        if (uper_put_length(po, 0, 0)) {
+            ASN__ENCODE_FAILED;
+        }
+        ASN__ENCODED_OK(er);
     }
 
 

--- a/skeletons/constr_SET_OF.c
+++ b/skeletons/constr_SET_OF.c
@@ -300,31 +300,38 @@ static int _el_addbytes(const void *buffer, size_t size, void *el_buf_ptr) {
     el_buf->length += size;
     return 0;
 }
+
+static void assert_unused_bits(const struct _el_buffer* p) {
+    if(p->length) {
+        assert((p->buf[p->length-1] & ~(0xff << p->bits_unused)) == 0);
+    } else {
+        assert(p->bits_unused == 0);
+    }
+}
+
 static int _el_buf_cmp(const void *ap, const void *bp) {
     const struct _el_buffer *a = (const struct _el_buffer *)ap;
     const struct _el_buffer *b = (const struct _el_buffer *)bp;
     size_t common_len;
-    int ret;
+    int ret = 0;
 
     if(a->length < b->length)
         common_len = a->length;
     else
         common_len = b->length;
 
-    ret = memcmp(a->buf, b->buf, common_len);
+    //constr_SET_OF.c:315:13: runtime error: null pointer passed as argument 1, which is declared to never be null
+    if (a->buf && b->buf) {
+        ret = memcmp(a->buf, b->buf, common_len);
+    }
     if(ret == 0) {
         if(a->length < b->length)
             ret = -1;
         else if(a->length > b->length)
             ret = 1;
         /* Ignore unused bits. */
-        if(a->length) {
-            assert((a->buf[a->length-1] & ~(0xff << a->bits_unused)) == 0);
-            assert((b->buf[b->length-1] & ~(0xff << b->bits_unused)) == 0);
-        } else {
-            assert(a->bits_unused == 0);
-            assert(b->bits_unused == 0);
-        }
+        assert_unused_bits(a);
+        assert_unused_bits(b);
     }
 
     return ret;

--- a/skeletons/constr_SET_OF.c
+++ b/skeletons/constr_SET_OF.c
@@ -1054,7 +1054,7 @@ SET_OF_encode_uper(const asn_TYPE_descriptor_t *td,
 
     for(encoded_edx = 0; (ssize_t)encoded_edx < list->count;) {
         ssize_t may_encode;
-        ssize_t edx;
+        size_t edx;
         int need_eom = 0;
 
         if(ct && ct->effective_bits >= 0) {
@@ -1065,7 +1065,7 @@ SET_OF_encode_uper(const asn_TYPE_descriptor_t *td,
             if(may_encode < 0) ASN__ENCODE_FAILED;
         }
 
-        for(edx = 0; edx < may_encode; edx++) {
+        for(edx = encoded_edx; edx < encoded_edx + may_encode; edx++) {
             const struct _el_buffer *el = &encoded_els[edx];
             if(asn_put_many_bits(po, el->buf,
                                  (8 * el->length) - el->bits_unused) < 0) {

